docs/tutorials/training_to_inference.md

```diff
@@ -0,0 +1,201 @@
+    1.  Add the `name` parameter to every input operator.
+    1.  If you’d like to use the [Model Autoconfiguration](https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/model_configuration.html#auto-generated-model-configuration) of DALI model:
+        1. Add `ndim` parameter to every input operator,
+        1. Add `dtype` parameter to every input operator,
+        1. Add `output_ndim` parameter to the `@pipeline_def`,
+        1. Add `output_dtype` parameter to the `@pipeline_def`.
+1. Create a [Model Repository](https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/model_repository.html).
+    1. Set up your Deep Learning model. Description of this step is outside of the scope of this tutorial.
+    1. Create a directory for the DALI model.
+    1. Insert the DALI pipeline definition (prepared in the **Step 1**) into the `dali.py` file in the version directory.
+    1. Create a [configuration file](https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/model_configuration.html) (unnecessary, if you are using Model Autoconfiguration for DALI Backend). Make sure that the names of the inputs in the model configuration match the names of the input operators assigned in **Step 1**.
+    1. Combine the Deep Learning and DALI models using [model ensemble](https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/architecture.html#ensemble-models) or [BLS script](https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/index.html?highlight=business%20logic%20scripting).
+1. Run the Triton server and send some requests to it.
+
+Enough with the theory. Let's go to practice.
+
+## Practice
+
+### ➔ 1: Adjusting DALI Pipeline
+
+As mentioned earlier, the validation pipeline usually reflects the inference preprocessing better. Here's one used in 
+our EfficientNet example:
+
+```python
+@pipeline_def
+def validation_pipe(data_dir, interpolation, image_size, image_crop, output_layout, rank=0,
+                    world_size=1):
+    jpegs, label = fn.readers.file(name="Reader", file_root=data_dir, shard_id=rank,
+                                   num_shards=world_size, random_shuffle=False, pad_last_batch=True)
+
+    images = fn.decoders.image(jpegs, device="mixed", output_type=types.RGB)
+
+    images = fn.resize(images, resize_shorter=image_size, interp_type=interpolation,
+                       antialias=False)
+
+    output = fn.crop_mirror_normalize(images, dtype=types.FLOAT, output_layout=output_layout,
+                                      crop=(image_crop, image_crop),
+                                      mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
+                                      std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+    return output, label
+```
+
+We're adjusting it by swapping the Reader with ExternalSource and adding a `name` parameter to it. Since in our example 
```

Comment:
```suggestion
We're adjusting it by swapping the reader with external_source and adding a `name` parameter to it. Since in our example 
```

just a suggestion. PascalCase names seem kinda outdated

Reply:
or even just `external source`

Reply:
Done